components/SignUpComponent.js

```diff
@@ -0,0 +1,69 @@
+import HomePage from "../pageobjects/BasePage";
+import { expect } from "@playwright/test";
+
+export default class SignUpComponent extends HomePage {
+  constructor(page) {
+    super(page);
+    this.button = this.page.locator(".hero-descriptor button");
+    this.modal = this.page.locator('[role="document"]');
+    this.name = this.modal.locator("#signupName");
+    this.lastName = this.modal.locator("#signupLastName");
+    this.email = this.modal.locator("#signupEmail");
+    this.password = this.modal.locator("#signupPassword");
+    this.repeatPassword = this.modal.locator("#signupRepeatPassword");
+    this.submit = this.modal.getByRole("button", {
+      name: "Register",
+    });
+    this.errorMessage = this.modal.locator(".invalid-feedback p");
+  }
+
+  async fillRegistrationForm(userName, userLastName, userEmail, password) {
+    await this.name.fill(userName);
+    await this.lastName.fill(userLastName);
+    await this.email.fill(`aqa+${userEmail}`);
+    await this.password.fill(password);
+    await this.repeatPassword.fill(password);
+    await this.submit.click();
```

Comment:
method with name fill should only fill and no other actions inside. How should I know there is a click inside?